pkg/controller/component/lifecycle/kbagent.go

```diff
@@ -390,3 +343,44 @@ func (a *kbagent) formatError(lfa lifecycleAction, rsp proto.ActionResponse) err
 		return wrapError(err)
 	}
 }
+
+func SelectTargetPods(pods []*corev1.Pod, pod *corev1.Pod, spec *appsv1.Action) ([]*corev1.Pod, error) {
+	if spec.Exec == nil || len(spec.Exec.TargetPodSelector) == 0 {
+		return []*corev1.Pod{pod}, nil
+	}
+
+	anyPod := func() []*corev1.Pod {
+		i := rand.Int() % len(pods)
+		return []*corev1.Pod{pods[i]}
```

Comment:
The controller does not check the availability of target Pods, and leaving it to the Addon and Action to choose, because the controller has no way of knowing which pods meet the availability requirement for a specific action.